src/plugins/WebOfTrust/util/SynchronousDelayedBackgroundJob.java

```diff
@@ -0,0 +1,231 @@
+        this.job = job;
+        this.name = name;
+        this.defaultDelay = delay;
+    }
+
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most the default
+     * delay, or as soon as the currently running job finishes, whichever comes last.
+     * If this background job was terminating or has terminated prior to the invocation of this
+     * method, this method has no effect and returns immediately.
+     * This method waits for the first job to start after the trigger to finish before returning,
+     * except when this method is called from the background job itself, in that case this method
+     * does not wait for the job to finish (otherwise a deadlock would be the result).
+     * If the calling thread is interrupted while waiting for the job to finish execution, this
+     * method will return as soon as possible, without waiting for the job to finish, and sets the
+     * thread interruption flag.
+     */
+    @Override
+    public synchronized void triggerExecution() {
+        triggerExecution(defaultDelay);
+    }
+
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most the given
+     * delay, or as soon as the currently running job finishes, whichever comes last. Negative
+     * delays are treated as to zero delays.
+     * If this background job was terminating or has terminated prior to the invocation of this
+     * method, this method has no effect and returns immediately.
+     * This method waits for the first job to start after the trigger to finish before returning,
+     * except when this method is called from the background job itself, in that case this method
+     * does not wait for the job to finish (otherwise a deadlock would be the result).
+     * If the calling thread is interrupted while waiting for the job to finish execution, this
+     * method will return as soon as possible, without waiting for the job to finish, and sets the
+     * thread interruption flag.
+     * @param delay the maximum trigger aggregation delay in milliseconds
+     */
+    @Override
+    public synchronized void triggerExecution(long delay) {
+        if (isTerminating) {
+            return;
+        }
+        triggerExecutionAsynchronously(delay);
+        // When called from the job thread, don't wait for the job to finish: this would cause a
+        // deadlock.
+        if (runningJobThread == Thread.currentThread()) {
+            return;
+        }
+        // Wait for the relevant execution to complete.
+        long waitForRunCount;
+        if (runningJobThread == null) {
+            // Wait for next run to finish
+            waitForRunCount = runCount + 1;
+        } else {
+            // Wait for next run after current run to finish
+            waitForRunCount = runCount + 2;
+        }
+        while (runCount < waitForRunCount) {
+            try {
+                wait();
```

Comment:
> Make it internally use a TickerDelayedBackgroundJob for its whole affairs instead of the current homebrew code.

Hmm before doing this first make sure that it's even possible to obtain a Ticker in unit tests. WOT sets it Tickers to null in unit test because it cannot obtain them because it has no PluginRespirator. Maybe it would be possible to manually construct them though: Part of the reason why WOT sets them to null is that the results of the delayed computation would be irrelevant since the unit tests only test immediate results.